src/py42/clients/settings/__init__.py

```diff
@@ -0,0 +1,199 @@
+from py42._compat import string_type
+from py42.clients.settings._converters import bool_to_str
+from py42.clients.settings._converters import no_conversion
+from py42.clients.settings._converters import str_to_bool
+
+
+def set_val(d, keys, value):
+    """Helper for setting nested values from a dict based on a list of keys."""
+    d = get_val(d, keys[:-1])
+    d[keys[-1]] = value
+
+
+def get_val(d, keys):
+    """Helper for getting nested values from a dict based on a list of keys."""
+    for key in keys:
```

Comment:
What is going on here?  Why not just set `d = d[keys[-1]]` ?

Reply:
Because there can be more than one level of nesting. This allows us to define each setting property with just the list of keys as the "location" of the value we're looking to get/set: 

```
backup_status_email_frequency_days = SettingProperty(
        name="backup_status_email_frequency_days",
        location=["settings", "serviceBackupConfig", "backupStatusEmailFreqInMinutes"],
        get_converter=minutes_to_days,
        set_converter=days_to_minutes,
    )
```

Reply:
Oh I see now

Reply:
It reminds me of the Ruby `dig` method: https://tiagoamaro.com.br/2016/08/27/ruby-2-3-dig/